MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -2071,12 +2083,58 @@ private string GetFixedSizeScalingFilter(Video3DFormat? threedFormat, int reques
             var isVaapiH264Encoder = outputVideoCodec.IndexOf("h264_vaapi", StringComparison.OrdinalIgnoreCase) != -1;
             var isQsvH264Encoder = outputVideoCodec.IndexOf("h264_qsv", StringComparison.OrdinalIgnoreCase) != -1;
             var isNvdecH264Decoder = videoDecoder.IndexOf("h264_cuvid", StringComparison.OrdinalIgnoreCase) != -1;
+            var isNvdecHevcDecoder = videoDecoder.IndexOf("hevc_cuvid", StringComparison.OrdinalIgnoreCase) != -1;
             var isLibX264Encoder = outputVideoCodec.IndexOf("libx264", StringComparison.OrdinalIgnoreCase) != -1;
             var isLinux = RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
+            var isColorDepth10 = IsColorDepth10(state);
 
             var hasTextSubs = state.SubtitleStream != null && state.SubtitleStream.IsTextSubtitleStream && state.SubtitleDeliveryMethod == SubtitleDeliveryMethod.Encode;
             var hasGraphicalSubs = state.SubtitleStream != null && !state.SubtitleStream.IsTextSubtitleStream && state.SubtitleDeliveryMethod == SubtitleDeliveryMethod.Encode;
 
+            // Currently only with the use of NVENC decoder can we get a decent performance.
+            // Currently only the HEVC/H265 format is supported.
+            // NVIDIA Pascal and Turing or higher are recommended.
+            if (isNvdecHevcDecoder && isColorDepth10
+                && _mediaEncoder.SupportsHwaccel("opencl")
+                && options.EnableTonemapping
+                && !string.IsNullOrEmpty(videoStream.VideoRange)
+                && videoStream.VideoRange.Contains("HDR", StringComparison.OrdinalIgnoreCase))
+            {
+                var parameters = "tonemap_opencl=format=nv12:primaries=bt709:transfer=bt709:matrix=bt709:tonemap={0}:desat={1}:threshold={2}:peak={3}";
```

Comment:
For the CPU ones at least (as I wrote in the other comment) scaling them first to floating point numbers stop some artifacting

Reply:
I haven't notice any annoying artifacts on this gpu accelerated filter.
The scaling is done with nvidia nvdec decoder. `-c:v hevc_cuvid -resize 1920x1080`
However, other decoders cannot do this, resulting in better performance of tonemap on nvidia gpu.

Reply:
I meant scaling the values with something like `zscale` en `format=gbrpf32le` bijvoorbeeld.

Reply:
I see. But this means that the new bottleneck will come from the cpu.

Reply:
If this opencl filter does not clip or give weird integer match artifacts I think it's fine without. zscale should be fairly quick on the CPU.